src/main/java/org/candlepin/insights/inventory/kafka/KafkaEnabledInventoryService.java

```diff
@@ -43,12 +42,12 @@
     private final KafkaTemplate<String, HostOperationMessage> producer;
     private final String hostIngressTopic;
 
-    public KafkaEnabledInventoryService(InventoryServiceProperties inventoryServiceProperties,
+    public KafkaEnabledInventoryService(String hostIngressTopic, int hostStaleOffset,
```

Comment:
Honestly, I'm not a huge fan of this change.  The `InventoryServiceProperty` class exists to encapsulate all the various settings that the service might need.  Unpacking them seems like something that should be done internally within implementations of the service (including the `DefaultInventoryService` for that matter).

Sending the `InventoryServiceProperties` object into the constructor does make unit testing a little annoying though.  We could get around this by using `@TestPropertySource` on a test class and sending in a string with our desired properties.  That annotation can only be applied to classes though so if we have different sets of properties, we'd need multiple inner classes or use something like ` TestPropertySourceUtils` to modify the values.

I'm going to mark this as "request changes" but if you feel strongly about the way you have it now, I'm not going to kick up a fuss.  It's just something that I thought you might want to consider.  If the big thing you don't like is messy tests, I'm willing to spend some time to see if I can come up with a better solution than creating `InventoryServiceProperty` objects for each test.

Reply:
I see your point. I'll see what I can do here.